Userland/Services/FileSystemAccessServer/DownloadWidget.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/URL.h>
+#include <LibCore/ElapsedTimer.h>
+#include <LibCore/FileStream.h>
+#include <LibGUI/Progressbar.h>
+#include <LibGUI/Widget.h>
+#include <LibProtocol/Request.h>
+
+namespace FileSystemAccessServer {
+
+class DownloadWidget final : public GUI::Widget {
+    C_OBJECT(DownloadWidget);
+
+public:
+    virtual ~DownloadWidget() override;
+
+private:
+    explicit DownloadWidget(const URL&, String const&);
```

Comment:
```suggestion
    DownloadWidget(URL const&, String const& destination_path);
```
It's not immediately obvious what the second parameters does, so let's give it a name. Also, `explicit` is only really necessary for single argument constructors.